core/engine/src/builtins/temporal/duration/mod.rs

```diff
@@ -929,94 +928,94 @@ pub(crate) fn to_temporal_partial_duration(
     };
 
     // 2. Let result be a new partial Duration Record with each field set to undefined.
-    let mut result = InnerDuration::partial();
+    let mut result = PartialDuration::default();
 
     // 3. NOTE: The following steps read properties and perform independent validation in alphabetical order.
     // 4. Let days be ? Get(temporalDurationLike, "days").
     let days = unknown_object.get(js_str!("days"), context)?;
     if !days.is_undefined() {
         // 5. If days is not undefined, set result.[[Days]] to ? ToIntegerIfIntegral(days).
-        result.set_days(f64::from(to_integer_if_integral(&days, context)?));
+        result.days = Some(FiniteF64::from(to_integer_if_integral(&days, context)?));
```

Comment:
suggestion: potentially use in-place `Option` operations.

There may be some benefit to use either `insert` or `get_or_insert` over " = Some(...) ". i.e. result.days.insert(FiniteF64::from(...));

Reply:
Yup done